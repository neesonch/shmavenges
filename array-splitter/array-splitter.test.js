const splitInto = require('./array-splitter.js');

const INPUT_ARRAY = ["Mercury", "Venus", "Earth", "Mars", "Jupiter", "Saturn", "Uranus", "Neptune", "Pluto (It's still a planet to me, dammit)"];

test('#splitArray into 0 arrays', () => {
  expect(splitInto(INPUT_ARRAY, 0)).toMatchSnapshot();
});

test('#splitArray into 1 array', () => {
  expect(splitInto(INPUT_ARRAY, 1)).toMatchSnapshot();
});

test('#splitArray into multiple arrays', () => {
  expect(splitInto(INPUT_ARRAY, 4)).toMatchSnapshot();
});

test('#splitArray into more arrays than original items', () => {
  expect(splitInto(INPUT_ARRAY, 10)).toMatchSnapshot();
});