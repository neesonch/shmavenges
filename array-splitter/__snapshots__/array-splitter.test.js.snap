// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#splitArray into 0 arrays 1`] = `Array []`;

exports[`#splitArray into 1 array 1`] = `
Array [
  Array [
    "Mercury",
    "Venus",
    "Earth",
    "Mars",
    "Jupiter",
    "Saturn",
    "Uranus",
    "Neptune",
    "Pluto (It's still a planet to me, dammit)",
  ],
]
`;

exports[`#splitArray into more arrays than original items 1`] = `
Array [
  Array [
    "Mercury",
  ],
  Array [
    "Venus",
  ],
  Array [
    "Earth",
  ],
  Array [
    "Mars",
  ],
  Array [
    "Jupiter",
  ],
  Array [
    "Saturn",
  ],
  Array [
    "Uranus",
  ],
  Array [
    "Neptune",
  ],
  Array [
    "Pluto (It's still a planet to me, dammit)",
  ],
  Array [],
]
`;

exports[`#splitArray into multiple arrays 1`] = `
Array [
  Array [
    "Mercury",
    "Jupiter",
    "Pluto (It's still a planet to me, dammit)",
  ],
  Array [
    "Venus",
    "Saturn",
  ],
  Array [
    "Earth",
    "Uranus",
  ],
  Array [
    "Mars",
    "Neptune",
  ],
]
`;
